### PR TITLE
Fix/revert to readcounter

### PR DESCRIPTION
**Pull request by lpoli**
### Changes
This PR reverts ReadMarker changes from read size to read counter.

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- 0chain: https://github.com/0chain/0chain/pull/1146
- gosdk: https://github.com/0chain/gosdk/pull/429
- system_test: https://github.com/0chain/system_test/pull/249
- zboxcli: https://github.com/0chain/zboxcli/pull/211
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/blobbercore/readmarker/worker.go [hunk 1]
```diff
@@ -6,113 +6,115 @@ import (
...
+			return
+		}
+		return // synced from blockchain, no redeeming needed
 	}
 
-	if err := tx.ExecuteSmartContract(transaction.STORAGE_CONTRACT_ADDRESS, transaction.READ_REDEEM,
-		string(rdRedeemBytes), 0); err != nil {
-		zLogger.Logger.Info("Failed submitting read redeem", zap.Error(err))
-		return common.NewErrorf("redeem_read_marker", "sending transaction: %v", err)
+	if latestRM.ReadCounter == rmEntity.LatestRM.ReadCounter {
```
Comment (peterlimg) [inline code/go/0chain.net/blobbercore/readmarker/worker.go]: Isn't this case is covered above in line 42? unless the latestRM.ReadCounter is 0, but is it necessary to run sync for it?

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/blobbercore/readmarker/worker.go (hunk 1 above)
Comment (lpoli) [reply code/go/0chain.net/blobbercore/readmarker/worker.go]: Yes in current code , absolutely no need to have it. It was previous logic and it I think it was done to skip redeem request if rmEntity.LatestRM.ReadCounter is zero.

**Review by peterlimg — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | fix/revert-to-readcounter  |
| zwalletcli  | staging  |
| 0chain  | fix/revert-to-readcounter  |
| blobber  | fix/revert-to-readcounter  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/revert-to-readcounter  |
| zboxcli  | fix/revert-to-readcounter  |
| zwalletcli  | staging  |
| 0chain  | fix/revert-to-readcounter  |
| blobber  | fix/revert-to-readcounter  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/revert-to-readcounter  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | fix/revert-to-readcounter  |
| zwalletcli  | staging  |
| 0chain  | fix/revert-to-readcounter  |
| blobber  | fix/revert-to-readcounter  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/revert-to-readcounter  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | fix/revert-to-readcounter  |
| zwalletcli  | staging  |
| 0chain  | fix/revert-to-readcounter  |
| blobber  | fix/revert-to-readcounter  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**